src/app/services/dialog.service.ts

```diff
@@ -1,20 +1,85 @@
 import { Injectable } from '@angular/core';
-import { MatDialog, MatDialogConfig } from '@angular/material/dialog';
+import { MatDialog, MatDialogConfig, MatDialogRef } from '@angular/material/dialog';
 import { UntilDestroy } from '@ngneat/until-destroy';
-import { Observable, of } from 'rxjs';
+import { Observable, Subject, of } from 'rxjs';
+import { ApiJobDirectory, ApiJobMethod } from 'app/interfaces/api/api-job-directory.interface';
 import {
   ConfirmOptions,
   ConfirmOptionsWithSecondaryCheckbox,
   DialogWithSecondaryCheckboxResult,
 } from 'app/interfaces/dialog.interface';
 import { ErrorReport } from 'app/interfaces/error-report.interface';
+import { Job, JobProgress } from 'app/interfaces/job.interface';
 import { ConfirmDialogComponent } from 'app/modules/common/dialog/confirm-dialog/confirm-dialog.component';
 import { ErrorDialogComponent } from 'app/modules/common/dialog/error-dialog/error-dialog.component';
 import { FullScreenDialogComponent } from 'app/modules/common/dialog/full-screen-dialog/full-screen-dialog.component';
 import { GeneralDialogComponent, GeneralDialogConfig } from 'app/modules/common/dialog/general-dialog/general-dialog.component';
 import { InfoDialogComponent } from 'app/modules/common/dialog/info-dialog/info-dialog.component';
+import { JobProgressDialogComponent, JobProgressDialogConfig } from 'app/modules/common/dialog/job-progress/job-progress-dialog.component';
 import { MultiErrorDialogComponent } from 'app/modules/common/dialog/multi-error-dialog/multi-error-dialog.component';
 
+export class JobProgressDialog {
+  private readonly onSuccess$ = new Subject<Job>();
+  private readonly onAbort$ = new Subject<Job>();
+  private readonly onFailure$ = new Subject<Job>();
+  private readonly _onProgress$ = new Subject<JobProgress>();
+
+  readonly afterSuccess$ = this.onSuccess$.asObservable();
+  readonly afterAbort$ = this.onAbort$.asObservable();
+  readonly afterFailure$ = this.onFailure$.asObservable();
+  readonly onProgress$ = this._onProgress$.asObservable();
```

Comment:
What's the difference between using `asObservable` vs returning a subject, which is also an observable?

Reply:
Subjects are allowed to emit values. The intention is to only expose buttons that are needed to the class's user. They don't need to do emits. So why expose that.